packages/ui/components/ui/segmented-picker.tsx

```diff
@@ -47,14 +62,28 @@ export const SegmentedPicker = <ValueType extends { toString: () => string }>({
           aria-checked={value === option.value}
           onClick={() => onChange(option.value)}
           className={cn(
-            'px-3 py-1 font-bold cursor-pointer',
-            index === 0 && 'rounded-l-sm',
-            index === options.length - 1 && 'rounded-r-sm',
-            value === option.value && 'bg-teal',
+            'font-bold cursor-pointer relative bg-background items-center justify-center flex',
+            getRoundedClasses(index, options.length, size),
+            size === 'md' && 'text-sm px-3 h-8',
+            size === 'lg' && 'px-4 h-10',
             value !== option.value && 'text-light-grey',
-            value !== option.value && 'bg-light-brown',
+            grow && 'grow',
           )}
         >
+          {value === option.value && (
+            <motion.div
+              layout
+              layoutId={layoutId}
```

Comment:
This `layoutId` prop is one of the crucial pieces. It lets Framer know that this is the div we're trying to match between segments